src/index.ts

```diff
@@ -1,7 +1,2 @@
-import configurePrim from './configurePrim'
-import defaultOptions from './defaultOptions'
-
 export { default as configurePrim } from './configurePrim'
 export { default as defaultOptions } from './defaultOptions'
-
-export const { PrimProvider, usePrim, primmed } = configurePrim(defaultOptions)
```

Comment:
The `defaultOptions` X `configurePrim`'s type inference appears to cross some threshold of TypeScript's `.d.ts` declaration generation. This will require more in-depth research to correct, but for now, luckily we're not dependent on the `defaultOptions`, and so removing these appears to generate valid type declarations.